sdk/openai/azure-ai-openai/src/main/java/com/azure/ai/openai/models/EmbeddingItem.java

```diff
@@ -9,25 +9,51 @@
 import com.azure.json.JsonSerializable;
 import com.azure.json.JsonToken;
 import com.azure.json.JsonWriter;
+
 import java.io.IOException;
 import java.util.List;
+import java.util.stream.Collectors;
+
 import static com.azure.ai.openai.implementation.EmbeddingsUtils.convertBase64ToFloatList;
+import static com.azure.ai.openai.implementation.EmbeddingsUtils.convertFloatListToBase64;
 
 /**
  * Representation of a single embeddings relatedness comparison.
  */
 @Immutable
 public final class EmbeddingItem implements JsonSerializable<EmbeddingItem> {
 
+    /*
+     * List of embeddings value for the input prompt. These represent a measurement of the
+     * vector-based relatedness of the provided input.
+     */
+    @Generated
+    private final List<Double> embedding;
+
     private final String embeddingBase64;
 
+    /**
+     * Creates an instance of EmbeddingItem class.
+     *
+     * @param embeddingBase64 the embedding value to set.
+     * @param promptIndex the promptIndex value to set.
+     */
+    private EmbeddingItem(String embeddingBase64, List<Double> embedding, int promptIndex) {
+        this.embeddingBase64 = embeddingBase64;
+        this.promptIndex = promptIndex;
+        this.embedding = embedding;
+    }
+
     /**
      * Get the embedding property: List of embeddings value for the input prompt. These represent a measurement of the
      * vector-based relatedness of the provided input.
      *
      * @return the embedding value.
      */
     public List<Float> getEmbedding() {
+        if (embedding != null) {
+            return embedding.stream().map(Double::floatValue).collect(Collectors.toList());
+        }
         return convertBase64ToFloatList(embeddingBase64);
```

Comment:
We don't want to call convert every time `getEmbedding` method is called. 

```suggestion
        embedding = convertBase64ToFloatList(embeddingBase64);
        return embedding;
```